plugins/importexport/medra/filter/ArticleMedraXmlFilter.inc.php

```diff
@@ -354,6 +354,10 @@ function createContributorNode($doc, $author, $objectLocalePrecedence) {
 		$invertedPersonName = $author->getFullName(false, true);
 		assert(!empty($invertedPersonName));
 		$contributorNode->appendChild($node = $doc->createElementNS($deployment->getNamespace(), 'PersonNameInverted', htmlspecialchars($invertedPersonName, ENT_COMPAT, 'UTF-8')));
+		// Key names (mandatory)
+		$keyNames = $author->getFullName(false);
+		assert(!empty($keyNames));
+		$contributorNode->appendChild($node = $doc->createElementNS($deployment->getNamespace(), 'KeyNames', htmlspecialchars($keyNames, ENT_COMPAT, 'UTF-8')));		
```

Comment:
`$keyNames` will be an array here, correct? This looks broken to me -- passing the array through `htmlspecialchars` and into `createElementNS` directly.

Reply:
Hello my dear, how are you? Do you have news? it's URGENT to perform update of mEDRA plugin. Could you please let me know?
Thanks!!!